drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,454 @@
+		return_mask |= RTC_ALARM_TIME_MASK_YEAR;
+	}
+
+	*mask = return_mask;
+
+	return 0;
+}
+
+static int nxp_irtc_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	struct nxp_irtc_data *data = dev->data;
+	int ret = data->is_alarm_pending;
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	data->is_alarm_pending = 0;
+
+	return ret;
+}
+
+static int nxp_irtc_alarm_set_callback(const struct device *dev, uint16_t id,
+				       rtc_alarm_callback callback, void *user_data)
+{
+	struct nxp_irtc_data *data = dev->data;
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	uint32_t key = irq_lock();
+
+	data->alarm_callback = callback;
+	data->alarm_user_data = user_data;
+	data->is_alarm_pending = 1;
+
+	irq_unlock(key);
+
+	return 0;
+}
+
+#endif /* CONFIG_RTC_ALARM */
+#if defined(CONFIG_RTC_UPDATE) || defined(__DOXYGEN__)
+
+static int nxp_irtc_update_set_callback(const struct device *dev, rtc_update_callback callback,
+					void *user_data)
+{
+	return -ENOTSUP;
+}
+
+#endif /* CONFIG_RTC_UPDATE */
+#if defined(CONFIG_RTC_CALIBRATION) || defined(__DOXYGEN__)
+
+static int nxp_irtc_set_calibration(const struct device *dev, int32_t calibration)
+{
+	return -ENOTSUP;
+}
+
+static int nxp_irtc_get_calibration(const struct device *dev, int32_t *calibration)
+{
+	return -ENOTSUP;
+}
+
+#endif /* CONFIG_RTC_CALIBRATION */
+
+static int nxp_irtc_init(const struct device *dev)
+{
+	const struct nxp_irtc_config *config = dev->config;
+	RTC_Type *irtc_reg = config->base;
+
+	nxp_irtc_unlock_registers(irtc_reg);
+
+	/* set the control register bits */
+	irtc_reg->CTRL = RTC_CTRL_CLK_SEL(config->clock_src) |
+			       RTC_CTRL_CLKO_DIS(!config->is_output_clock_enabled);
+
+	config->irq_config_func(dev);
+
+	return 0;
+}
+
+static void nxp_irtc_isr(const struct device *dev)
+{
+#ifdef CONFIG_RTC_ALARM
+	const struct nxp_irtc_config *config = dev->config;
+	RTC_Type *irtc_reg = config->base;
+	struct nxp_irtc_data *data = dev->data;
+	uint32_t key = irq_lock();
+
+	nxp_irtc_unlock_registers(irtc_reg);
+	irtc_reg->ISR = irtc_reg->ISR;
+
+	if (data->alarm_callback) {
+		data->alarm_callback(dev, 0, data->alarm_user_data);
+		data->is_alarm_pending = 0;
+	} else {
+		data->is_alarm_pending = 1;
+	}
+	irq_unlock(key);
+#endif /* CONFIG_RTC_ALARM */
+}
+
+static const struct rtc_driver_api rtc_nxp_irtc_driver_api = {
+	.set_time = nxp_irtc_set_time,
+	.get_time = nxp_irtc_get_time,
+#if defined(CONFIG_RTC_ALARM) || defined(__DOXYGEN__)
+	.alarm_get_supported_fields = nxp_irtc_alarm_get_supported_fields,
+	.alarm_set_time = nxp_irtc_alarm_set_time,
+	.alarm_get_time = nxp_irtc_alarm_get_time,
+	.alarm_is_pending = nxp_irtc_alarm_is_pending,
+	.alarm_set_callback = nxp_irtc_alarm_set_callback,
+#endif /* CONFIG_RTC_ALARM */
+#if defined(CONFIG_RTC_UPDATE) || defined(__DOXYGEN__)
+	.update_set_callback = nxp_irtc_update_set_callback,
+#endif /* CONFIG_RTC_UPDATE */
+#if defined(CONFIG_RTC_CALIBRATION) || defined(__DOXYGEN__)
```

Comment:
Remove the `defined(__DOXYGEN__)` from the `#if`, its in the `rtc.h` header for docs generation, its not used in drivers :)

Reply:
These will be removed on the next update.